git-cola: added .desktop file

agvantibo-again opened the pull request:
git-cola was missing its bubbles icon and was
displayed inconsistently in desktop environments


<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

agvantibo-again committed:
git-cola: added .desktop file and icon

git-cola was missing its bubbles icon and was
displayed inconsistently in desktop environments

agvantibo-again commented:
Should I introduce a linked issue? Furthermore, should I try to fetch some kind of icon in-package? git-cola seems to be pretty common, present in breeze and qogir and [probably] adwaita

pbsds commented:
I think it makes sense to install an icon. I can see multiple candidates in src.

agvantibo-again commented:
> I think it makes sense to install an icon. I can see multiple candidates in src.

I can't find them. Could you please give me a pointer? There is a /share directory in the source, but there's no icon to be found. Desktop items aplenty though; it would probably make sense to copy those instead of remaking them arbitrarily like I tried to

agvantibo-again commented:
> I think it makes sense to install an icon. I can see multiple candidates in src.

I've added the cola/icons/git-cola.svg icon for all feasible icon sizes and rewrote desktopItems to use the files supplied by src, they seem to work alright.

pbsds commented:
```
git-cola> magick: no images found for operation `-resize' at CLI arg 3 @ error/operation.c/CLIOption/5481.
```

agvantibo-again commented:
> ```
> git-cola> magick: no images found for operation `-resize' at CLI arg 3 @ error/operation.c/CLIOption/5481.
> ```
My bad; the migration from `convert` to `magick` introduced a restriction on the order of the arguments that I didn't catch and verify; this builds properly now
